### PR TITLE
fix: Copy command buffers

### PR DESCRIPTION
**Pull request by eriksvedang**

### CODE REVIEW

File: src/template/csharp/unity/unity-replace-entity.go [hunk 1]
```diff
@@ -49,6 +49,17 @@ namespace {{.Namespace}}
 			}
 		{{end}}
 		{{end}}
+
+        // Command buffers
+        {{range .Commands}}
+            if (!entityManager.HasComponent<{{.EngineName}}>(source)) {
```
Comment (rpadrela) [inline src/template/csharp/unity/unity-replace-entity.go:55]: Using `source` here as argument but using `destination` to `AddBuffer()` below. Is this correct?

Comment (eriksvedang) [reply src/template/csharp/unity/unity-replace-entity.go:55]: I'm pretty sure it is. It checks if the thing we're copying from has the buffer and then adds it to the thing we're copying to. Arguably it should copy the contents of the buffer also...

Comment (rpadrela) [reply src/template/csharp/unity/unity-replace-entity.go:55]: Ah! Nevermind, we are adding components to the destination that exist in the source, isn't it?

Comment (rpadrela) [reply src/template/csharp/unity/unity-replace-entity.go:55]: > I'm pretty sure it is. It checks if the thing we're copying from has the buffer and then adds it to the thing we're copying to. Arguably it should copy the contents of the buffer also...

Right...

**Comment by eriksvedang** [comment]
Turns out it was hard to copy the contents of the buffers because it invalidates some references which makes the code unburstable. I noticed I got the logic wrong though, but that should be fixed now. @rpadrela

**Comment by rpadrela** [comment]
> Turns out it was hard to copy the contents of the buffers because it invalidates some references which makes the code unburstable. I noticed I got the logic wrong though, but that should be fixed now. @rpadrela

What are the consequences of not copying the contents? Is it ok not to?

**Comment by eriksvedang** [comment]
@rpadrela In theory it could mean that you drop commands that are sent to an entity as it is being created.

**Review by rpadrela — APPROVED** [review]
Ship it!

**Pull request merged**